command/agent/cache/lease_cache.go

```diff
@@ -1031,14 +1044,17 @@ func (c *LeaseCache) restoreLeases(leases [][]byte) error {
 		}
 
 		if err := c.restoreLeaseRenewCtx(newIndex); err != nil {
-			return err
+			errors = multierror.Append(errors, err)
+			continue
 		}
 		if err := c.db.Set(newIndex); err != nil {
-			return err
+			errors = multierror.Append(errors, err)
+			continue
 		}
 		c.logger.Trace("restored lease", "id", newIndex.ID, "path", newIndex.RequestPath)
 	}
-	return nil
+
```

Comment:
nit: extra newline seems to differ from the style of the other functions in this change